src/PIL/Image.py

```diff
@@ -702,7 +702,7 @@ def __array_interface__(self):
                     pass
                 else:
                     if parse_version(numpy.__version__) < parse_version("1.23"):
-                        warnings.warn(e)
+                        warnings.warn(str(e))
```

Comment:
How did this look before/after?

Reply:
[Without this change](https://github.com/radarhere/Pillow/commit/4005a52b54d1cd1ae7897c4120af570f4db0f7f8), I get https://github.com/radarhere/Pillow/actions/runs/8296217068/job/22704816536#step:11:2665
> TypeError: Warning must be str or Warning, got OSError('image file is truncated (8 bytes not processed)') (type OSError)